libraries/AP_GPS/GPS_detect_state.h

```diff
@@ -68,3 +68,24 @@ struct SBP_detect_state {
     uint16_t crc_so_far;
     uint16_t crc;
 };
+
+struct SBF_detect_state {
+    enum
+    {
+        PREAMBLE1 = 0,
+        PREAMBLE2,
+        CRC1,
+        CRC2,
+        BLOCKID1,
+        BLOCKID2,
+        LENGTH1,
+        LENGTH2,
+        DATA
+    } sbf_state;
+    uint16_t preamble;
+    uint16_t crc;
+    uint16_t blockid;
+    uint16_t length;
+    uint8_t data[256];
```

Comment:
what do we need 256 bytes of data for in _detect() ? Doesn't this just waste memory? It doesn't seem to be used, and this is static data, so we pay for it on all boards